test/jdk/com/sun/java/swing/plaf/gtk/TestFileChooserSingleDirectorySelection.java

```diff
@@ -61,19 +63,36 @@ public static void main(String[] args) throws Exception {
 
         try {
             // create test directory
-            String tmpDir = System.getProperty("user.home");
-            testDir = new File[1];
-            testDir[0] = new File(tmpDir, "testDir");
-            if (!testDir[0].exists())
-                testDir[0].mkdir();
-            testDir[0].deleteOnExit();
-
-            // create temporary files inside testDir
-            tempFile = new File[5];
+            String tmpDir = System.getProperty("java.io.tmpdir");
+
+            // Create a test directory that contains only folders
+            testDir = new File(tmpDir, "testDir");
+            if (!testDir.exists()) {
+                testDir.mkdir();
+            }
+            testDir.deleteOnExit();
+
+            // create sub directories inside test directory
+            SubDirs = new File[5];
+            for (int i = 0; i < 5; ++i) {
+                SubDirs[i] = new File(testDir, "subDir_" + (i+1));
+                SubDirs[i].mkdir();
+                SubDirs[i].deleteOnExit();
+            }
+
+            // Create a test directory that contains only files
+            testFile = new File(tmpDir, "testFile");
+            if (!testFile.exists()) {
+                testFile.mkdir();
+            }
+            testFile.deleteOnExit();
+
+            // create temporary files inside testFile
+            subFiles = new File[5];
             for (int i = 0; i < 5; ++i) {
-                tempFile[i] = File.createTempFile("temp", ".txt",
-                        new File(testDir[0].getAbsolutePath()));
-                tempFile[i].deleteOnExit();
+                subFiles[i] = File.createTempFile("subFiles_" + (i+1),
+                        ".txt", new File(testFile.getAbsolutePath()));
+                subFiles[i].deleteOnExit();
             }
         } catch (Exception e) {
             e.printStackTrace();
```

Comment:
The exception should've been propagated: if an exception is thrown, test setup code failed to prepare the expected environment — the test would likely fail, then let it fail quickly and report the real reason.